app/controllers/pensions/incomeFromOverseasPensions/PensionOverseasIncomeStatus.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.pensions.incomeFromOverseasPensions
+
+import config.{AppConfig, ErrorHandler}
+import controllers.predicates.AuthorisedAction
+import forms.YesNoForm
+import models.User
+import models.mongo.PensionsCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.Clock
+import views.html.pensions.incomeFromOverseasPensions.IncomeFromOverseasPensionsView
+import controllers.pensions.incomeFromOverseasPensions.routes.PensionOverseasIncomeStatus
+import controllers.pensions.routes.PensionsSummaryController
+
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class PensionOverseasIncomeStatus @Inject()(authAction: AuthorisedAction,
+                                            incomeFromOverseasPensionsView: IncomeFromOverseasPensionsView,
+                                            pensionSessionService: PensionSessionService,
+                                            errorHandler: ErrorHandler
+                                           )(implicit val mcc: MessagesControllerComponents,
+                                             appConfig: AppConfig,
+                                             clock: Clock,
+                                             ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport {
+
+  def yesNoForm(user: User): Form[Boolean] = YesNoForm.yesNoForm(
+    missingInputError = s"incomeFromOverseasPensions.incomeFromOverseasPension.error.noEntry.${if (user.isAgent) "agent" else "individual"}"
+  )
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    pensionSessionService.getPensionSessionData(taxYear, request.user).flatMap {
+      case Left(_) => Future.successful(errorHandler.handleError(INTERNAL_SERVER_ERROR))
+      case Right(optPensionUserData) => optPensionUserData match {
+        case Some(data) =>
+          data.pensions.incomeFromOverseasPensionsViewModel.paymentsFromOverseasPensionsQuestion match {
```

Comment:
This could be a map instead.
```
          Future.successful(data.pensions.incomeFromOverseasPensionsViewModel.paymentsFromOverseasPensionsQuestion
            .map(yesNo => Ok(incomeFromOverseasPensionsView(yesNoForm(request.user).fill(yesNo), taxYear)))
            .getOrElse(Ok(incomeFromOverseasPensionsView(yesNoForm(request.user), taxYear))))
```
And you could remove future successful if you use map instead of flatMap

Reply:
That is a separate issue you, mind.

But I do agree with Salman about using a map instead of the pattern matching on Some / None. Not idiomatic.

Reply:
Usually the convention follows a more functional style, using a map is more consistent for accessing monads. Futures, Lists, Eithers are mapped on. You could argue a getOrElse can return a different type than what was declared in the option. If you wanted to avoid that you can fold on the option:
`myOption.fold(defaultValue)(a=>a) ` <- Note this was changed in scala 3 where a different type would now be accepted.

Reply:
I've opted for this as a slight middle ground. I think it's a bit more readable 
`          val form = data.pensions.incomeFromOverseasPensionsViewModel.paymentsFromOverseasPensionsQuestion.fold(yesNoForm(request.user)
          )(yesNoForm(request.user).fill(_))
          Future.successful(Ok(incomeFromOverseasPensionsView(form, taxYear)))`

Reply:
Readable perhaps (in the eye of the beholder), but less expected than a `getOrElse`; and inconsistent with the direction elsewhere in the code.

But I'll step away from this particular matter, and leave it to Salman.

Reply:
`          val form =
            data.pensions.incomeFromOverseasPensionsViewModel.paymentsFromOverseasPensionsQuestion.fold(yesNoForm(request.user)
            )(yesNoForm(request.user).fill(_))
          Future.successful(Ok(incomeFromOverseasPensionsView(form, taxYear)))`

Reply:
I would prefer getOrElse but I can't fault the suggested implementation.

Reply:
Ok cool, i guess it's all preferences. I'll push this up :)